packages/cw721/src/execute.rs

```diff
@@ -0,0 +1,678 @@
+use cosmwasm_std::{
+    Addr, Api, BankMsg, Binary, Coin, CustomMsg, Deps, DepsMut, Empty, Env, MessageInfo, Response,
+    StdResult, Storage,
+};
+use cw_ownable::{none_or, Action, Ownership, OwnershipError, OwnershipStore};
+use cw_storage_plus::Item;
+use cw_utils::Expiration;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+use crate::{
+    error::Cw721ContractError,
+    msg::{Cw721ExecuteMsg, Cw721InstantiateMsg, Cw721MigrateMsg},
+    receiver::Cw721ReceiveMsg,
+    state::{CollectionInfo, Cw721Config, DefaultOptionMetadataExtension, NftInfo, MINTER},
+    Approval,
+};
+
+pub trait Cw721Execute<
+    // Metadata defined in NftInfo (used for mint).
+    TMetadataExtension,
+    // Defines for `CosmosMsg::Custom<T>` in response. Barely used, so `Empty` can be used.
+    TCustomResponseMessage,
+    // Message passed for updating metadata.
+    TMetadataExtensionMsg,
+> where
+    TMetadataExtension: Serialize + DeserializeOwned + Clone,
+    TCustomResponseMessage: CustomMsg,
+    TMetadataExtensionMsg: CustomMsg,
+{
+    fn instantiate(
+        &self,
+        deps: DepsMut,
+        _env: Env,
+        info: MessageInfo,
+        msg: Cw721InstantiateMsg,
+        contract_name: &str,
+        contract_version: &str,
+    ) -> Result<Response<TCustomResponseMessage>, Cw721ContractError> {
+        cw2::set_contract_version(deps.storage, contract_name, contract_version)?;
```

Comment:
@shanev look here. Like `Cw721Execute` in cw721 package. It is still an interface/trait. Only difference here is that it provides a default implementation like here in `instantiate()`. This way contract just can implement `Cw721Execute` and in case it custom logic is required, contract can implement its own logic - but in 99% of the cases it can just use default logic.

Reply:
We did the same for `cw-ics721`, by providing default logic - like many other projects: https://github.com/public-awesome/cw-ics721/blob/main/packages/ics721/src/execute.rs#L36-L48